aiozk/recipes/shared_lock.py

```diff
@@ -1,27 +1,48 @@
-from aiozk import exc, Deadline
-
 from .base_lock import BaseLock
 
 
 class SharedLock(BaseLock):
-    async def acquire_read(self, timeout=None):
-        deadline = Deadline(timeout)
-        result = None
-        while not result:
+    class LockProxy:
```

Comment:
Probably we can make this code cleaner if we allow passing `wait_with_party` parameter as the parameter of `Lock` and just use two locks.

Reply:
OK. Reusing Lock class looks more simple.
But some extra keyword arguments should be added to it.
I will try it.